src/pip/_internal/network/xmlrpc.py

```diff
@@ -34,7 +31,8 @@ def __init__(self, index_url, session, use_datetime=False):
         self._session = session
 
     def request(self, host, handler, request_body, verbose=False):
-        # type: (str, str, Dict[str, str], bool) -> None
+        # type: (_HostType, str, bytes, bool) -> Tuple[_Marshallable, ...]
+        assert isinstance(host, str)
```

Comment:
Why can’t `host` be annotated as `str` but need this extra dance?

Reply:
I imagine this is for compatibility with the base class? Still tho, it'd be useful to know how mypy failed without the signature change.

Reply:
If we annotate this as str, mypy emits the following error:

```
src/pip/_internal/network/xmlrpc.py:33: error:
Argument 1 of "request" is incompatible with supertype "Transport"; supertype defines the argument type as "Union[Tuple[str, Dict[str, str]], str]"
        def request(self, host, handler, requ...

src/pip/_internal/network/xmlrpc.py:33: note: This violates the Liskov substitution principle
src/pip/_internal/network/xmlrpc.py:33: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
```

OTOH, without the assert, mypy emits the following error:

```
src/pip/_internal/network/xmlrpc.py:36: error:
Value of type variable "AnyStr" of "urlunparse" cannot be "Sequence[Collection[str]]"
            url = urllib.parse.urlunparse(par...
```

This is because `_HostType` is defined as:

https://github.com/python/typeshed/blob/a0cc2f537ddd7d6fec9ee3fd46bef6994e44aadd/stdlib/3/xmlrpc/client.pyi#L18

```
_HostType = Union[Tuple[str, Dict[str, str]], str]
```
However, this class doesn't support the tuple form. One way to tell mypy to ignore it is to assert that it is of a specific type.